### PR TITLE
Port foreman_os_default_templates to apypie

### PR DESCRIPTION
**Pull request by mdellweg**

### CODE REVIEW

File: plugins/modules/foreman_os_default_template.py [hunk 1]
```diff
@@ -126,35 +103,30 @@ def main():
...
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    entity_dict['operatingsystem'] = find_operating_system_by_title(module, entity_dict['operatingsystem'])
-    entity_dict['template_kind'] = find_entities_by_name(
-        TemplateKind, [entity_dict['template_kind']], module)[0]
+    search = 'title~"{}"'.format(entity_dict['operatingsystem'])
+    entity_dict['operatingsystem'] = module.find_resource('operatingsystems', search, thin=True)
```
Comment (evgeni) [inline plugins/modules/foreman_os_default_template.py]: Given my comment in https://github.com/theforeman/foreman-ansible-modules/pull/366#discussion_r306633638, I really think we should move that to a helper now ;)

File: plugins/modules/foreman_os_default_template.py [hunk 2]
```diff
@@ -80,42 +80,19 @@
...
-        'provisioning_template': 'provisioning_template',
-    }
-    result = {}
-    for key, value in name_map.items():
-        if key in entity_dict:
-            result[value] = entity_dict[key]
-    return result
+name_map = {
+    'operatingsystem': 'operatingsystem',
+    'template_kind': 'template_kind',
```
Comment (evgeni) [inline plugins/modules/foreman_os_default_template.py]: I "like" that we use `template_kind` here, but `kind` in the provisioning template. Time to consolidate? :)

**Comment by evgeni** [comment]
needs rebase (prov template was merged) and rerecord :)

File: plugins/modules/foreman_os_default_template.py (hunk 1 above)
Comment (mdellweg) [reply plugins/modules/foreman_os_default_template.py]: How about a fuzzy Flag?

Comment (evgeni) [reply plugins/modules/foreman_os_default_template.py]: Sounds good

Comment (bagasse) [reply plugins/modules/foreman_os_default_template.py]: I given my comment in https://github.com/theforeman/foreman-ansible-modules/pull/366#discussion_r306678268 on this too :)

**Pull request merged**